app/controllers/concerns/api/import_puppetclasses_common_controller.rb

```diff
@@ -26,7 +27,7 @@ def import_puppetclasses
     if params[:except].present?
       kinds = params[:except].split(',')
       kinds.each do |kind|
-        @changed[kind] = {} if ["new", "obsolete", "updated", "ignored"].include?(kind)
+        @changed[kind] = {} if ALLOWED_KINDS.include?(kind)
```

Comment:
Does this need to be a constant or can this be a variable that is not defined in the repeated block?

Reply:
Variable outside of loop is probably good enough, constant is the extreme solution I agree. If you insist, I can rewrite those to variables which is still an array allocation per function call tho.

Reply:
I don't have a strong opinion here. I guess if I had to decide, I'd do the variable.

Reply:
Yeah but this was not autocorrection rule and it's few cycles faster sooooo... ;-)

Reply:
The difference is probably that the global variable makes the overall memory usage higher since it's always allocated. There is also the risk that something mutates the variable leading to unexpected results. I'd go with the variable inside the method.

Reply:
Main issue with constants for me is, we should care more about their naming and placement.